Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -60,7 +60,7 @@ public BlockStoreBehavior(ConcurrentChain chain, IBlockRepository blockRepositor
             Guard.NotNull(chain, nameof(chain));
             Guard.NotNull(blockRepository, nameof(blockRepository));
             Guard.NotNull(blockStoreCache, nameof(blockStoreCache));
-            Guard.NotNull(blockStoreCache, nameof(logger));
+            Guard.NotNull(blockStoreCache, nameof(this.logger));
```

Comment:
this line was wrong and still is wrong, it checks blockStoreCache, but it most likely was supposed to check the loggerFactory argument

Reply:
Fixed.